src/nix/derivation-instantiate.md

```diff
@@ -0,0 +1,39 @@
+R""(
+
+# Name
+
+`nix derivation instantiate` - instantiate store derivations
+
+# Synopsis
+
+`nix derivation instantiate`
+  [`--out-link` *link prefix*]
+  [`--json`]
+  [`--no-link`]
+  *installables…*
+
+# Description
+
+The command `nix derivation instantiate` produces [store derivation]s from
+installables. Each top-level expression should evaluate to a derivation, a list
+of derivations, or a set of derivations. The paths of the resulting store
+derivations are printed on standard output.
+
+[store derivation]: @docroot@/glossary.md#gloss-store-derivation
+
+# Options
+
+- `--out-link` *link prefix*
+
+  The prefix used for gc roots.
+
+- `--no-link`
+
+  Do not create garbage collector roots for the generated store derivations.
+
+- `--json`
+
+  Dump a JSON object whose keys are the generated store derivations instread of
```

Comment:
```suggestion
  Dump a JSON object whose keys are the generated store derivations instead of
```

Reply:
It would be preferable to return a list, so that multiple installables can be distinguished by their position, reflecting that of the command line.

This is also how `nix build` behaves.

Nice to have:
It would be possible to add an `outputs` field as well, with unbuilt paths. Only input-addressed output paths and fixed-output path are known ahead of time (not ca or impure), so they should be in `.drv^out` form or a JSON representation of that. Let's save that for a follow-up PR.

Thought (no action needed):
It also touches on https://github.com/NixOS/nix/issues/6507, but it's probably fine for this only work on derivations and not package attrsets (which aren't supported in the CLI yet).

Reply:
The "defaultOutput" name would be very useful.
I think the output should contain roughly what we store in the eval cache database. These two things are similar.